### DISCUSSION TITLE
opening a viewer directly

### DISCUSSION THREAD

**Discussion by beroal** [roddhjav/apparmor.d#526] [Q&A]
I am working on a profile for AMule. It's a GUI file downloader. It has a command to preview files. When this command is executed, it starts, for example, a video player (VLC), not `xdg-open`. Do I understand it correctly that the `child-open` profile is useless in this case? How can I allow this command without allowing to execute everything?

**Comment by roddhjav** [comment]
Well known program of identical purpose are managed here: https://github.com/roddhjav/apparmor.d/blob/main/apparmor.d/tunables/multiarch.d/paths

So you can use any `@{*_path}` variable directly that is needed for a given program.

**Reply by beroal** [reply]
AMule can download files of any type. I'm afraid it reimplements `xdg-open`, so what it runs for preview depends on what is installed on the user's computer. If VLC is installed, AMule tries to run VLC. If not, AMule runs nothing.

Luckily, it's perfectly usable without this preview command.
